### PR TITLE
Add Version to golangci

### PR DESCRIPTION
**Pull request by marbar3778**
- added version to golangci because of recent update

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -46,3 +46,6 @@ linters:
 #       - wrapperFunc
 #       - commentFormatting # https://github.com/go-critic/go-critic/issues/755
 
+service:
+  golangci-lint-version: 1.17.x # use the fixed version to not introduce new linters unexpectedly
+  prepare:
```
Comment (marbar3778) [inline .golangci.yml]: doing this now, will go through and update to most recent and clean up according to the added linters in the coming days.

File: rpc/core/pipe.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ var (
 	// objects
 	pubKey           crypto.PubKey
 	genDoc           *types.GenesisDoc // cache the genesis structure
-	addrBook         p2p.AddrBook
+	addrBook         p2p.AddrBook      // nolint: unused
```
Comment (melekes) [inline rpc/core/pipe.go]: unused? can be removed if so

File: .golangci.yml [hunk 3]
```diff
@@ -46,3 +46,6 @@ linters:
 #       - wrapperFunc
 #       - commentFormatting # https://github.com/go-critic/go-critic/issues/755
 
+service:
+  golangci-lint-version: 1.17.x # use the fixed version to not introduce new linters unexpectedly
```
Comment (melekes) [inline .golangci.yml]: how often new linters are added? feel like we'll never be updating lint version if it will require us manually changing the version

Comment (marbar3778) [reply .golangci.yml]: i can also disable the new linters for now and work through them this week

**Pull request merged**